katcp/resource_client.py

```diff
@@ -669,7 +676,7 @@ def _add_sensors(self, sensor_keys):
         sensor_instances = yield sensor_instance_fut
         # Store KATCPSensor instances in self.sensor
         added_names = []
-        for s_name, s_obj in sensor_instances.items():
+        for s_name, s_obj in list(sensor_instances.items()):
```

Comment:
2to3 tends to be very conservative and makes these conversions to give identical behaviour to the old code.

In a lot of cases what you probably really want is
```python
for s_name, s_obj in six.iteritems(sensor_instances)
```
to iterate over the original rather than making a copy, but it needs to be done carefully: if the loop mutates the dictionary then it's not equivalent.